File: leetcode/java/two-sum.java
class Solution {
  public int[] twoSum(int[] nums, int target) {
    java.util.HashMap<Integer, Integer> m=new HashMap<Integer, Integer>(nums.length);
    for(int i=0;i<nums.length;i++) {
      if(m.containsKey(target-nums[i]))return new int[] {m.get(target-nums[i]), i};
      m.put(nums[i],i);
    }
  }
}

File: leetcode/algorithms/TwoSum.cs
using System;
using System.Collections.Generic;
class Solution 
{
  public int[] TwoSum(int[] nums, int target) 
  {
    Dictionary<int, int> TargetMinusNumsElement = new Dictionary<int, int>();
    for (int i = 0; i < nums.Length; i++)
    {
      if(!TargetMinusNumsElement.ContainsKey(nums[i]))
        TargetMinusNumsElement[target-nums[i]] = i; 
      else
      {
        return new int[2] {i, TargetMinusNumsElement[nums[i]]};
      }
    }
  }
}
